### PR TITLE
move to separate groups

### PR DESCRIPTION
**Pull request by xh3b4sd**
We want to have non-cluster resources to be in a non-cluster group. This PR adds a new `core` group to put ingress and cert resources in there.

### CODE REVIEW

**Comment by xh3b4sd** [comment]
Following the steps described in https://github.com/giantswarm/apiextensions/pull/10 I still fail to generate new clients successfully. I feel like there should be some additional magic in the `register.go` and we should make sure this is also documented then. 

```
$ go test ./...
?   	github.com/giantswarm/apiextensions/pkg/apis/cluster/v1alpha1	[no test files]
?   	github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1	[no test files]
# github.com/giantswarm/apiextensions/pkg/clientset/versioned/scheme
pkg/clientset/versioned/scheme/register.go:53:2: undefined: "github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1".AddToScheme
```

**Comment by kopiczko** [comment]
Yes, I forgot about this thing. Will add.

File: pkg/apis/cluster/v1alpha1/register.go [hunk 1]
```diff
@@ -27,9 +27,15 @@ var (
 // Adds the list of known types to api.Scheme.
 func addKnownTypes(scheme *runtime.Scheme) error {
 	scheme.AddKnownTypes(SchemeGroupVersion,
+		&AWS{},
+		&AWSList{},
 		&Azure{},
 		&AzureList{},
+		&KVM{},
+		&KVMList{},
```
Comment (xh3b4sd) [inline pkg/apis/cluster/v1alpha1/register.go]: This was missing. We should make sure this is documented as well.

File: pkg/apis/cluster/v1alpha1/zz_generated.deepcopy.go [hunk 2]
```diff
@@ -873,283 +873,6 @@ func (in *AzureVirtualNetwork) DeepCopy() *AzureVirtualNetwork {
 	return out
 }
 
-// DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
-func (in *Cert) DeepCopyInto(out *Cert) {
```
Comment (xh3b4sd) [inline pkg/apis/cluster/v1alpha1/zz_generated.deepcopy.go]: This moved to the core group now.

File: pkg/apis/core/v1alpha1/doc.go [hunk 3]
```diff
@@ -0,0 +1,4 @@
+// +k8s:deepcopy-gen=package,register
+
+// +groupName=core.giantswarm.io
+package v1alpha1
```
Comment (xh3b4sd) [inline pkg/apis/core/v1alpha1/doc.go]: This was missing as well. We should make sure this is documented.

File: scripts/gen.sh [hunk 4]
```diff
@@ -18,4 +18,4 @@ cd ${dir}/../vendor/k8s.io/code-generator && ./generate-groups.sh \
     all \
     github.com/giantswarm/apiextensions/pkg \
     github.com/giantswarm/apiextensions/pkg/apis \
-    "cluster:v1alpha1"
+    "cluster:v1alpha1 core:v1alpha1"
```
Comment (xh3b4sd) [inline scripts/gen.sh]: Here is the new group registered for generation.

**Comment by xh3b4sd** [comment]
Fishing for LGTM.

File: pkg/clientset/versioned/clientset.go [hunk 5]
```diff
@@ -29,13 +30,17 @@ type Interface interface {
 	ClusterV1alpha1() clusterv1alpha1.ClusterV1alpha1Interface
 	// Deprecated: please explicitly pick a version if possible.
 	Cluster() clusterv1alpha1.ClusterV1alpha1Interface
+	CoreV1alpha1() corev1alpha1.CoreV1alpha1Interface
+	// Deprecated: please explicitly pick a version if possible.
+	Core() corev1alpha1.CoreV1alpha1Interface
```
Comment (JosephSalisbury) [inline pkg/clientset/versioned/clientset.go]: lol at the autogeneration creating deprecated functions

Comment (xh3b4sd) [reply pkg/clientset/versioned/clientset.go]: I can imagine this is for transitioning. Good to be aware this should not be used anymore.

**Comment by xh3b4sd** [comment]
@kopiczko lets make sure we document the learnings here in your PR you created to improve the docs.

**Pull request merged**